payjoin/src/persist/mod.rs

```diff
@@ -0,0 +1,30 @@
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+pub trait PersistableValue: Serialize + DeserializeOwned + Sized + Clone {}
```

Comment:
Persistable value should have a method to get its own key. 
This way we can remove the K and get the token from the persistable value.